src/Evolution/Systems/Burgers/FiniteDifference/BoundaryGhostData.hpp

```diff
@@ -0,0 +1,196 @@
+#include "Evolution/Systems/Burgers/FiniteDifference/Reconstructor.hpp"
+#include "Evolution/Systems/Burgers/Tags.hpp"
+#include "NumericalAlgorithms/Spectral/Mesh.hpp"
+#include "Utilities/ErrorHandling/Assert.hpp"
+#include "Utilities/Gsl.hpp"
+#include "Utilities/PrettyType.hpp"
+#include "Utilities/TMPL.hpp"
+
+namespace Burgers::fd {
+/*!
+ * \brief Computes finite difference ghost data for external boundary
+ * conditions.
+ *
+ */
+struct BoundaryGhostData {
+  template <typename FdBoundaryConditionHelper, typename DbTagsList,
+            typename... FdBoundaryConditionArgsTags>
+  // A helper function for calling fd_ghost() of BoundaryCondition subclasses
+  static void apply_subcell_boundary_condition_impl(
+      FdBoundaryConditionHelper& fd_boundary_condition_helper,
+      const gsl::not_null<db::DataBox<DbTagsList>*>& box,
+      tmpl::list<FdBoundaryConditionArgsTags...>) {
+    return fd_boundary_condition_helper(
+        db::get<FdBoundaryConditionArgsTags>(*box)...);
+  }
+
+  template <typename DbTagsList>
+  static void apply(const gsl::not_null<db::DataBox<DbTagsList>*> box,
+                    const Element<1>& element,
+                    const Burgers::fd::Reconstructor& reconstructor) {
+    const ::Domain<1>& domain = db::get<domain::Tags::Domain<1>>(*box);
+    const auto& external_boundary_condition =
+        domain.blocks()[element.id().block_id()].external_boundary_conditions();
+
+    const Mesh<1> subcell_mesh =
+        db::get<evolution::dg::subcell::Tags::Mesh<1>>(*box);
+
+    const size_t ghost_zone_size{reconstructor.ghost_zone_size()};
+
+    for (const auto& direction : Direction<1>::all_directions()) {
+      if (not(element.neighbors().contains(direction))) {
+        // If a direction is not in elements.neighbors(), then the element must
+        // be at the external boundary
+        ASSERT(element.external_boundaries().count(direction),
+               "Element contains a direction which is neither neighbor nor "
+               "external boundary");
+        // Also check if the pointer is not null. This will catch an error that
+        // the pointer has a correct type but not pointing anything
+        ASSERT(external_boundary_condition.at(direction) != nullptr, "");
+        const auto& boundary_condition =
+            *external_boundary_condition.at(direction);
+
+        const size_t num_face_pts{
+            subcell_mesh.extents().slice_away(direction.dimension()).product()};
+
+        // a Variables object to store the computed FD ghost data
+        Variables<tmpl::list<Burgers::Tags::U>> ghost_data_vars{
+            ghost_zone_size * num_face_pts};
+
+        // We don't need to care about boundary ghost data when using the
+        // periodic condition, so exclude it from the tmpl list here
+        using derived_boundary_conditions_for_subcell = tmpl::remove<
+            Burgers::BoundaryConditions::standard_boundary_conditions,
```

Comment:
You should get the boundary condition list out of the metavariables. See line 672 of `src/Evolution/DiscontinuousGalerkin/Actions/BoundaryConditionsImpl.hpp`